src/main/webapp/gallery.js

```diff
@@ -1,13 +1,34 @@
 // Played around with the following packages: masonry.pkgd.js, imagesloaded.pkgd.js
 // May need to use different libraries due to commercial use requirement
+// Masonry is released under the MIT license: https://desandro.mit-license.org/
 
 // Initialize Masonry
+console.log("initializing masonry");
 var $grid = $('.grid').masonry({
-  itemSelector: '.grid-item',
-  percentPosition: true,
-  columnWidth: '.grid-sizer'
+    itemSelector: '.grid-item',
+    percentPosition: true,
+    columnWidth: '.grid-sizer',
+    horizontalOrder: true
 });
-// Layout Masonry after each image finishes loading
-$grid.imagesLoaded().progress( function() {
-  $grid.masonry();
-});  
+
+fetchSpeciesData();
```

Comment:
Since you have body onload call this function, I don't think you need to call it here?

Reply:
I think I took it out from the onload call, so it gets called here instead